app/models/patient.rb

```diff
@@ -93,10 +93,14 @@ def self.pledged_status_summary(line)
                          resolved_without_fund: [false, nil])
                   .where.not(fund_pledge: [0, nil])
 
-    patients = base.where(pledge_sent_at: start_of_period..)
-                   .or(base.where(fund_pledged_at: start_of_period..))
-                   .order(fund_pledged_at: :asc)
-                   .select(*plucked_attrs)
+    if Config.use_appointment_date_for_budget_bar_hard_pledges?
+      patients = base.where(appointment_date: start_of_period..)
+    else
+      patients = base.where(pledge_sent_at: start_of_period..)
+    end
+    patients = patients.or(base.where(fund_pledged_at: start_of_period..))
```

Comment:
I think this line will catch patients who are pledged this period with an appointment in later periods, so we need to add change the budget bar itself to not mark them as hard pledged.

![image](https://github.com/user-attachments/assets/699ad54c-7c54-49f6-a60e-87d1e7cd4ac7)

Actually as i'm thinking it over, how should we handle the soft pledges here? should soft pledges appear in the current period (when the pledge is created) or the one tied to the appointment date?